### PR TITLE
Format configuration.md with code fences to avoid render issues

### PR DESCRIPTION
**Pull request by joaofnfernandes**
This page is being pulled into docs.docker.com and sometimes not rendering properly. I think Jekyll doesn't really like mixing markdown and html syntax, so I've changed the code samples to use fences instead of spaces. That seems to make Jekyll happier and infer better markdown and html blocks.

The last table in the document was also not closing and html tag which was causing rendering issues too.

The docs are currently pulling from `releases/2.5`, so if you want I can also submit PRs against `releases/2.5` and `releases/2.6`.

### CODE REVIEW

**Comment by joaofnfernandes** [comment]
@dmcgowan can you check if this makes sense?

File: docs/configuration.md [hunk 1]
```diff
@@ -48,215 +54,217 @@ This section lists all the registry configuration options. Some options in
...
+          password: password
+          insecure: true
+        from: sender@example.com
+        to:
+          - errors@example.com
+loglevel: debug # deprecated: use "log"
+storage:
+  filesystem:
+    rootdirectory: /var/lib/registry
+maxthreads: 100
```
Comment (dmcgowan) [inline docs/configuration.md]: spacing was messed up before with tabs, can you align this with rootdirectory

**Comment by joaofnfernandes** [comment]
Should be fixed

**Comment by stevvooe** [comment]
LGTM

Please submit PRs for the other branches.

**Pull request merged**